dts/bindings/regulator/regulator-gpio.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 EPAM Systems
+# SPDX-License-Identifier: Apache-2.0
+
+description: GPIO-controlled voltage/current of regulators
+
+include:
+  - name: base.yaml
+  - name: regulator.yaml
+    property-allowlist:
+      - regulator-name
+      - regulator-min-microvolt
+      - regulator-max-microvolt
+
+compatible: "regulator-gpio"
+
+properties:
+  regulator-name:
+    required: true
+
+  gpios:
+    type: phandle-array
+    required: true
+    description: |
+      GPIO to use to switch voltage.
+
+  gpios-state:
+    type: int
+    description: |
+      This array provides the state of GPIO pin set when requesting it from
+      the gpio controller.
+
+  states:
+    type: array
+    description: Selection of available voltages provided by this regulator and
+      matching GPIO configurations to achieve them. If there are no states in
+      the "states" array, use a fixed regulator instead. First value in an array
+      item is voltage in microvolts and the second is GPIO group state value.
+
+  enable-gpios:
```

Comment:
this is not supported by the driver, please add it

Reply:
This property doesn't have a state array at all and doesn't support a few GPIOs. In other words, it works in the same way as the `enable-gpios` property inside the `regulator-fixed` Zephyr driver.

Reply:
@aasinclair pls, let me know if you want to see the state array for this property

Reply:
I've enhanced description of this field and add example of usage of the gpio regulator
I hope, I understood your request correctly